src/util/prolog/prolog.test.ts

```diff
@@ -0,0 +1,296 @@
+
+		test("One Variable Two Results", async () => {
+
+				const session = new PrologSession("likes(lisa, bob). likes(lisa, dieter).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X).")).getResultArray();
+				expect(results).toStrictEqual([["bob"], ["dieter"], ["false"]]);
+
+		});
+
+		test("Two Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl). likes(lisa, karl, bob).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl"], ["karl", "bob"], ["false"]]);
+
+		});
+
+		test("Three Variables Two Results", async () => {
+				const session = new PrologSession("likes(lisa, bob, karl, friedrich). likes(lisa, bob, friedrich, karl).", 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z).")).getResultArray();
+
+				expect(results).toStrictEqual([["bob", "karl", "friedrich"], ["bob", "friedrich", "karl"], ["false"]]);
+
+		});
+
+		test("Four Variables Two Results", async () => {
+				const session = new PrologSession(`
+					likes(lisa, bob, karl, friedrich, dieter).
+					likes(lisa, bob, karl, dieter, friedrich).
+					`, 1000);
+				const results = (await session.executeQuery("likes(lisa, X, Y, Z, A).")).getResultArray();
+
+				expect(results).toStrictEqual(
+					[["bob", "karl", "friedrich", "dieter"], ["bob", "karl", "dieter", "friedrich"], ["false"]],
+				);
+
+		});
+
+		test("Cut Operator Maximum Function", async () => {
+
+				const session = new PrologSession("max(X,Y,Y)  :-  X  =<  Y, !. max(X, Y, X).", 1000);
+				const results = (await session.executeQuery("max(3, 7, X).")).getResultArray();
+				expect(results).toStrictEqual([["7"], ["false"]]);
+
+		});
+
+		test("Cut Operator Endless Loop", async () => {
+
+				const session = new PrologSession("likes(lisa, bob, karl). likes(X, Y, Z):- likes(X, Z, Y), !.", 1000);
+				const results = (await session.executeQuery("likes(lisa, Y, Z).")).getResultArray();
+				expect(results).toStrictEqual([["bob", "karl"], ["karl", "bob"], ["false"]]);
+
+		});
+
+		test("Only Write", async () => {
+
+				const session = new PrologSession("write('Hello').", 1000);
+				const results = (await session.executeQuery("")).getResultArray();
+				expect(results).toStrictEqual([["false"]]);
+
+		});
+
+		test("Write Function and Query", async () => {
+
+				const session = new PrologSession("write('Hello'). likes(lisa, bob).", 1000);
+				const results = (await session.executeQuery("likes(lisa, bob).")).getResultArray();
+				expect(results).toStrictEqual([["true"], ["false"]]);
+
+		});
+
+		test("Truth Value (True) Perfomance", async () => {
+				const session = new PrologSession("likes(bob, lisa).", 1000);
+				const results = (await session.executeQuery("likes(bob, lisa).")).getResultArray();
+				expect(results).toStrictEqual([["true"], ["false"]]);
+		});
+
+		/* No longer needed with the introduction of maps.
```

Comment:
There shouldn't be any commented-out code anywhere - if you no longer need it, just delete it. If you ever want to look at this code again it's always there for you in the `git` history.